annotation-processor/src/main/java/com/hubspot/annotation/processor/ModernizerAnnotationProcessor.java

```diff
@@ -0,0 +1,151 @@
+        List<String> annotatedClasses = new ArrayList<String>();
+        for (Element element : roundEnv.getElementsAnnotatedWith(annotation)) {
+            List<String> warnings = Arrays.asList(
+                    element.getAnnotation(SuppressWarnings.class).value());
+            if (warnings.contains("modernizer")) {
+                if (element.getKind().isClass()) {
+                    annotatedClasses.add(getClassHeader(element));
+                }
+            }
+        }
+        return annotatedClasses;
+    }
+
+    private void makeAnnotatedClassesFile(List<String> annotatedClasses) {
+        FileWriter writer = null;
+        try {
+            File dir = new File(getFilePath(), "modernizer");
+            if (!dir.exists()) {
+                dir.mkdir();
+            }
+            File file = new File(dir.getPath(),
+                "modernizer-ignore-annotated-classes.txt");
+            writer = new FileWriter(file);
+            for (String element : annotatedClasses) {
+                writer.write(element + "\n");
+            }
+            writer.close();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        } finally {
+            if (writer != null) {
+                try {
+                    writer.close();
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        }
+    }
+
+    private String getFilePath() {
+        try {
+            FileObject fileObjectToGetPath =
+                processingEnv.getFiler().createResource(
+                    StandardLocation.CLASS_OUTPUT,
+                    "",
+                    "dummy-file.txt",
+                    null);
+            String path = fileObjectToGetPath.getName();
+            fileObjectToGetPath.delete();
+            while (!path.endsWith("target")) {
+                path = path.substring(0, path.lastIndexOf("/"));
+            }
+            return path;
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private String getClassHeader(Element classElement) {
+        String packageName = processingEnv
+            .getElementUtils()
+            .getPackageOf(classElement)
+            .getQualifiedName()
+            .toString()
+            .replace('.', '/');
+        String classHeader;
+        if (!packageName.isEmpty()) {
+            classHeader =
+                packageName + "/" + getFullClassName(classElement);
+        } else {
+            classHeader = getFullClassName(classElement);
+        }
+        return classHeader.replace("$", "\\$") + "(\\$.+)?";
+    }
+
+    private String getFullClassName(Element classElement) {
+        List<String> parentClasses = new ArrayList<String>();
+        Element enclosingElement = classElement.getEnclosingElement();
+        while (enclosingElement.getKind().isClass()) {
```

Comment:
Should we also check that `enclosingElement != null`? If the class doesn't have a package my guess is that the current code would NPE.

Reply:
If the class doesn't have a package, this method won't be called. The logic for this is written in `getClassHeader`

Reply:
It looks like this is always called in `getClassHeader`.  I see it in both branches of the if-else.

Reply:
You're right. I should check for it being null. Will update it.